includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,423 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\CLI\Plugin_Check_Command
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\CLI;
+
+use WordPress\Plugin_Check\Checker\CLI_Runner;
+use WordPress\Plugin_Check\Plugin_Context;
+use Exception;
+use WP_CLI\ExitException;
+
+/**
+ * Plugin check command.
+ */
+class Plugin_Check_Command {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @var Plugin_Context
+	 */
+	protected $plugin_context;
+
+	/**
+	 * Output format type.
+	 *
+	 * @var string[]
+	 */
+	protected $output_formats = array(
+		'table',
+		'csv',
+		'json',
+	);
+
+	/**
+	 * Check flags.
+	 *
+	 * @var string[]
+	 */
+	protected $check_flags = array(
+		'stable',
+		'beta',
+		'experimental',
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Plugin_Context $plugin_context Plugin context.
+	 */
+	public function __construct( $plugin_context ) {
+		$this->plugin_context = $plugin_context;
+	}
+
+	/**
+	 * Run plugin check.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<plugin>]
+	 * : The plugin to check. Plugin name.
+	 *
+	 * [--checks]
+	 * : Only runs checks provided as an argument in comma-separated values, e.g. enqueued-scripts, escaping. Otherwise runs all checks.
+	 *
+	 * [--flag]
+	 * : Limit the checks being executed according to their flags, e.g. stable, beta or experimental. Default is stable.
+	 * ---
+	 * default: stable
+	 * options:
+	 *   - stable
+	 *   - beta
+	 *   - experimental
+	 * ---
+	 *
+	 * [--format]
+	 * : Format to display the results. Options are table, csv, and json. The default will be a table.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 * ---
+	 *
+	 * [--fields]
+	 * : Limit displayed results to a subset of fields provided.
+	 *
+	 * [--ignore-warnings]
+	 * : Limit displayed results to exclude warnings.
+	 *
+	 * [--ignore-errors]
+	 * : Limit displayed results to exclude errors.
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   wp plugin check akismet
+	 *   wp plugin check akismet --check=escaping
+	 *   wp plugin check akismet --format=json
+	 *
+	 * @subcommand check
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @throws Exception Throws exception.
+	 */
+	public function check( $args, $assoc_args ) {
+
+		$assoc_args = $this->get_options( $assoc_args );
+
+		$this->get_plugin_from_args( $args );
```

Comment:
This doesn't do anything right now. We should probably put it into a variable, and then ensure that that plugin is actually being tested by comparing it to the plugin that the `CLI_Runner` contains.

But at this point we can't do that since that class doesn't expose the plugin tested anywhere. Let's just skip handling the plugin for now.

Reply:
Fixed this now, thanks!